process/scQueryProcessor.go

```diff
@@ -36,27 +38,35 @@ func NewSCQueryProcessor(proc Processor, pubKeyConverter core.PubkeyConverter) (
 }
 
 // ExecuteQuery resolves the request by sending the request to the right observer and replies back the answer
-func (scQueryProcessor *SCQueryProcessor) ExecuteQuery(query *data.SCQuery) (*vm.VMOutputApi, error) {
+func (scQueryProcessor *SCQueryProcessor) ExecuteQuery(query *data.SCQuery) (*vm.VMOutputApi, data.BlockInfo, error) {
 	addressBytes, err := scQueryProcessor.pubKeyConverter.Decode(query.ScAddress)
 	if err != nil {
-		return nil, err
+		return nil, data.BlockInfo{}, err
 	}
 
 	shardID, err := scQueryProcessor.proc.ComputeShardId(addressBytes)
 	if err != nil {
-		return nil, err
+		return nil, data.BlockInfo{}, err
 	}
 
 	observers, err := scQueryProcessor.proc.GetObservers(shardID)
 	if err != nil {
-		return nil, err
+		return nil, data.BlockInfo{}, err
 	}
 
 	for _, observer := range observers {
 		request := scQueryProcessor.createRequestFromQuery(query)
 		response := &data.ResponseVmValue{}
 
-		httpStatus, err := scQueryProcessor.proc.CallPostRestEndPoint(observer.Address, SCQueryServicePath, request, response)
+		path := scQueryServicePath
+		if len(query.BlockHash) > 0 {
+			path = fmt.Sprintf("%s%s%s", path, blockHashSuffix, hex.EncodeToString(query.BlockHash))
+		}
+		if query.BlockNonce.HasValue {
+			path = fmt.Sprintf("%s%s%d", path, blockNonceSuffix, query.BlockNonce.Value)
```

Comment:
if a user provides both coordinates, the URL will look like this: 
`observer:port/vm-values/query?blockHash=hash?blockNonce=37`
refactor to make it work with both coordinates as well

Reply:
fixed